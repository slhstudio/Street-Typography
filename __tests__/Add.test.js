import React from 'react';
import AddComponent from '../client/components/Add';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('<Add/>', () => {
  it ('renders and displays properly if there is no user', () => {
    const wrapper = shallow(<AddComponent isUser={false}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('p').exists()).toBe(true);
    // expect(wrapper.find('p').text()).toBe('If you would like to add photos, please log in.');
  });
  it ('renders and displays properly if there IS a user', () => {
    const wrapper = shallow(<AddComponent isUser={true}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('form').exists());
    // expect(wrapper.find('Input')).toHaveLength(5);
    // expect(wrapper.find('button').exists()).toBe(true);
    // expect(wrapper.find('button').text()).toBe('UPLOAD');
  });
  it('shows controlled form values correctly', () => {
    const wrapper = shallow(<AddComponent isUser={true}/>);
    wrapper.setState({ 
      notes : 'description goes here', 
      location : {
        address : 'New York, NY USA', 
        longitude : -74.0059728, 
        latitude : 40.7127753
      }
    });
    expect(toJSON(wrapper)).toMatchSnapshot();
    // const notesInput = wrapper.find('form').childAt(1);
    // expect(notesInput.props().value).toBe('description goes here');
    // const addressInput = wrapper.find('form').childAt(2);
    // expect(addressInput.props().value).toBe('New York, NY USA');
    // const lngInput = wrapper.find('form').childAt(3);
    // expect(lngInput.props().value).toBe(-74.0059728);
    // const latInput = wrapper.find('form').childAt(4);
    // expect(latInput.props().value).toBe(40.7127753);
  })
  it('handles conditional rendering correctly', () => {
    const wrapper = shallow(<AddComponent isUser={true}/>);
    wrapper.setState({ photoName : '' });
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('form').exists()).toBe(true);
    wrapper.setState({ photoName : 'xyz' });
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('Redirect').exists()).toBe(true);
  })
});