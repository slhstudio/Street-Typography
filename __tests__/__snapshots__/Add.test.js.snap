// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> handles conditional rendering correctly 1`] = `
<div>
  <Form
    onSubmit={[Function]}
  >
    <Input
      accept="image/gif, image/png, image/jpeg"
      name="image"
      onChange={[Function]}
      purpose="addForm"
      required={true}
      type="file"
    />
    <Input
      name="notes"
      onChange={[Function]}
      placeholder="Add some notes here"
      purpose="addForm"
      type="text"
      value=""
    />
    <Input
      name="address"
      onChange={[Function]}
      placeholder="Address (or closest guess)"
      purpose="addForm"
      required={true}
      type="text"
      value=""
    />
    <Input
      name="longitude"
      onChange={[Function]}
      placeholder="Longitude"
      required={true}
      type="hidden"
      value=""
    />
    <Input
      name="latitude"
      onChange={[Function]}
      placeholder="Latitude"
      required={true}
      type="hidden"
      value=""
    />
    <Button
      text="UPLOAD"
      type="submit"
    />
  </Form>
</div>
`;

exports[`<Add /> handles conditional rendering correctly 2`] = `
<Redirect
  push={false}
  to="photo/xyz"
/>
`;

exports[`<Add /> renders and displays properly if there IS a user 1`] = `
<div>
  <Form
    onSubmit={[Function]}
  >
    <Input
      accept="image/gif, image/png, image/jpeg"
      name="image"
      onChange={[Function]}
      purpose="addForm"
      required={true}
      type="file"
    />
    <Input
      name="notes"
      onChange={[Function]}
      placeholder="Add some notes here"
      purpose="addForm"
      type="text"
      value=""
    />
    <Input
      name="address"
      onChange={[Function]}
      placeholder="Address (or closest guess)"
      purpose="addForm"
      required={true}
      type="text"
      value=""
    />
    <Input
      name="longitude"
      onChange={[Function]}
      placeholder="Longitude"
      required={true}
      type="hidden"
      value=""
    />
    <Input
      name="latitude"
      onChange={[Function]}
      placeholder="Latitude"
      required={true}
      type="hidden"
      value=""
    />
    <Button
      text="UPLOAD"
      type="submit"
    />
  </Form>
</div>
`;

exports[`<Add /> renders and displays properly if there is no user 1`] = `
<div>
  <p>
    If you would like to add photos, please log in.
  </p>
</div>
`;

exports[`<Add /> shows controlled form values correctly 1`] = `
<div>
  <Form
    onSubmit={[Function]}
  >
    <Input
      accept="image/gif, image/png, image/jpeg"
      name="image"
      onChange={[Function]}
      purpose="addForm"
      required={true}
      type="file"
    />
    <Input
      name="notes"
      onChange={[Function]}
      placeholder="Add some notes here"
      purpose="addForm"
      type="text"
      value="description goes here"
    />
    <Input
      name="address"
      onChange={[Function]}
      placeholder="Address (or closest guess)"
      purpose="addForm"
      required={true}
      type="text"
      value="New York, NY USA"
    />
    <Input
      name="longitude"
      onChange={[Function]}
      placeholder="Longitude"
      required={true}
      type="hidden"
      value={-74.0059728}
    />
    <Input
      name="latitude"
      onChange={[Function]}
      placeholder="Latitude"
      required={true}
      type="hidden"
      value={40.7127753}
    />
    <Button
      text="UPLOAD"
      type="submit"
    />
  </Form>
</div>
`;
